### PR TITLE
ATO-153 Update dockerfile

### PR DESCRIPTION
**Pull request by CarlyG55**
## What?

Update dockerfile and implement docker compose
## Why?

Use the correct version of java, and source control running the stub from docker via docker compose

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,12 @@
-FROM gradle:jdk16@sha256:d31e12d105e332ec2ef1f31c20eac6d1467295487ac70e534e3c1d0ae4a0506e AS build
-COPY --chown=gradle:gradle . /home/gradle/src
+FROM gradle:7.6.3-jdk17 AS build
```
Comment (ethanmills) [inline Dockerfile:1]: It's best practice to pin a sha here

Comment (ethanmills) [inline Dockerfile:1]: Don't mind if it's in this change or another, but we should set up dependabot for docker

File: README.MD [hunk 2]
```diff
@@ -4,13 +4,12 @@
 This stub allows you to set various paramaters (see [here]("https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/")) and mock calls from an RP.
 
 ## Local Setup
-You will need to have the `di-authentication-api` repo locally (In the same parent directory by default)
+You'll need to set up environment variables to run locally. Make a copy of the `.env.template` file, and rename it `.env`.
+The stub is currently set up to connect to the build environment (as sandpit blocks requests from localhost). Once you 
+have a copy of `.env`, you'll need to go into the `build-client-registry` table in DynamoDB, and find the client with ClientName `di-auth-stub-relying-party-build`.
+Update the value of `CLIENT_ID` to the ClientID of this client.
```
Comment (ethanmills) [inline README.MD:10]: I'll try to chat to Dan about this to see if we can just drop this rule for sandpit

Comment (CarlyG55) [reply README.MD:10]: I believe we've rolled back this change now

**Review by ethanmills — APPROVED** [review]

**Pull request merged**